cs/src/core/RuntimeBonded.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Bond
+{
+    using System;
+
+    /// <summary>
+    /// Interface representing bonded payload of unknown type
+    /// </summary>
+    public interface IRuntimeBonded
+    {
+        /// <summary>
+        /// Deserialize an object from the IRuntimeBonded instance
+        /// </summary>
+        /// <returns>Deserialized object</returns>
+        object Deserialize();
+    }
+
```

Comment:
I thought that reusing the existing IBonded interfaces would have made it too easy to lose track of whether a bonded object is a "normal" one (i.e. a .NET class) or an instance of RuntimeObject. The method signatures would also make it would seem like a bonded RuntimeObject could be passed into the "normal" Serializer (or vice-versa), which would never work.

Making Deserialize generic seemed much less helpful than it is with the "normal" Deserializer, since the likely use case for all of this is in reflection-like scenarios where you don't know what type is being deserialized (and all calls would use "object" as the generic type argument, anyway).